module/os/linux/zfs/zfs_ioctl_os.c

```diff
@@ -148,6 +148,28 @@ zfsdev_ioctl(struct file *filp, unsigned cmd, unsigned long arg)
 
 }
 
+static int
+zfs_ioc_userns_attach(zfs_cmd_t *zc)
+{
+
+	if (zc == NULL)
+		return (SET_ERROR(EINVAL));
+
+	return (zone_dataset_attach(CRED(), zc->zc_name,
+	    (unsigned int)zc->zc_zoneid));
+}
+
+static int
+zfs_ioc_userns_detach(zfs_cmd_t *zc)
+{
+
+	if (zc == NULL)
+		return (SET_ERROR(EINVAL));
+
+	return (zone_dataset_detach(CRED(), zc->zc_name,
+	    (unsigned int)zc->zc_zoneid));
```

Comment:
Seems like we should also detach the namespace when destroying a dataset.

Reply:
We do not do this for FreeBSD jails (or illumos zones). Where would be the right place to put such a call?